app/models/fetcher_base.rb

```diff
@@ -0,0 +1,28 @@
+class FetcherBase
+  attr_reader :query, :topic
+
+  def initialize(query)
+    @query = query
+    @topic = :fetch_by_subject
```

Comment:
One thing I found a confusing coming back to this after a little while, was all this mutability with topic and to a lesser degree strategy. I wonder if it would be better to have `reducer.extract_fetcher` return a fetcher that is already set up how the reducer wants it, that is then given the query as an argument to `def extracts`. This would allow you to split it up into a `BySubjectFetcher` and a `ByUserFetcher`, which will help make the new `FetcherBase` do so more.

Quick sketch of this thought here: https://github.com/zooniverse/caesar/compare/bug-fix-pr-961...marten:mv-sketch-cleanup-fetcher?expand=1

Reply:
Agree - I originally though that this pattern was strange as well and mentioned it on the original PR, https://github.com/zooniverse/caesar/pull/961/files#r313807737

I prefer the version you've outlined, will look at incorporating / refactoring that once the review of this PR is in.